garak/resources/common.py

```diff
@@ -43,19 +45,25 @@
 
 
 def load_advbench(size: int = 0) -> pd.DataFrame:
-    advbench_base_path = (
-        garak._config.transient.package_dir
-        / "resources"
-        / "advbench"
-        / "harmful_behaviors.csv"
-    )
+    try:
+        advbench_base_path = data_path / "advbench" / "harmful_behaviors.csv"
+    except GarakException:
+        logging.info(
+            "No default harmful_behaviors.csv found download or cached file will be utilized."
+        )
+
+    # this utilizes the cache_dir however should this place the file in the data_dir now?
```

Comment:
This sounds right -- the file should be downloaded to `data_path / "advbench" / "harmful_behaviors.csv"`. We can probably do away with anything below. Just use `advbench_base_path` -> `advbench_path` and use that everywhere. Once you have it, we should be referencing it in a single location.

Reply:
Thanks, due to this comment, I think I see a bug here as well since as coded `advbench_path` is only defined if the file existed in data already access to `advbench_path.is_file()` would raise exceptions.

At this time I am thinking of `data_path` as being `read_only` for runtime code, meaning anything downloaded is currently place in `cache_dir`.  This class currently exposes the one location where support for user override instead of download is being enabled and I am not sure yet if downloads should mix data in the a user provided override path.

I will leave the comment and fix the guard for this PR.